### DISCUSSION TITLE
Cannot upgrade using Helm

### DISCUSSION THREAD

**Discussion by eric-aops** [strimzi/strimzi-kafka-operator#8014] [General]
**Describe the bug**
Upgrading from 0.32.0 to 0.33.0 fails with an error.
Previous upgrades worked like a charm.

**To Reproduce**
1. have 0.32.0 deployed and running
2. upgrade to version 0.33.0 with helm upgrade
3. Error

`Error: UPGRADE FAILED: rendered manifests contain a resource that already exists. Unable to continue with update: Lease "strimzi-cluster-operator" in namespace "strimzi" exists and cannot be imported into the current release: invalid ownership metadata; label validation error: missing key "app.kubernetes.io/managed-by": must be set to "Helm"; annotation validation error: missing key "meta.helm.sh/release-name": must be set to "strimzi"; annotation validation error: missing key "meta.helm.sh/release-namespace": must be set to "strimzi"
`

I tried manually editing the Lease resource but no joy.

**Expected behavior**
Expect to have a functional upgrade process.

**Environment (please complete the following information):**
 - Strimzi version: 0.32.0
 - Installation method: Helm chart
 - Kubernetes cluster: 1.23.14
 - Infrastructure: Google GKE

**YAML files and logs**

No specific configuration or changes, default Helm chart to deploy/upgrade the operator.

**Comment by scholzj** [comment]
The 0.33 Helm Chart contains the Lease resource. So you have to tell Helm to add it to the Helm installation.

**Reply by michael-barker** [reply]
We're also having this issue. What do you mean by telling Helm to add the file to the Helm installation?

**Reply by scholzj** [reply]
Check out https://cloud-native.slack.com/archives/CMH3Q3SNP/p1675088509172959 ... I'm not a Helm user, so no idea what other options are there to get Helm to recognize this file.

**Comment by eric-aops** [comment]
Hi, 

 

The error shown indicates the solution, if I would have read it good (RTFM) I wouldn’t have created this issue 😊. 

Just edit the Lease on the cluster:

 

kubectl edit Lease strimzi-cluster-operator

Add the two annotations and the label as indicated in the error, like below

  annotations:

    meta.helm.sh/release-name: strimzi #replace with your release-name

    meta.helm.sh/release-namespace: strimzi #replace with the namespace you deployed into

  labels:

    app.kubernetes.io/managed-by: Helm

 

Kind regards,

 

Eric Van Steenbergen

 

Phone: +356 7711 1767

Email: ***@***.***

 

From: Jakub Scholz ***@***.***>
Reply to: strimzi/strimzi-kafka-operator ***@***.***>
Date: Thursday, 2 February 2023 at 17:16
To: strimzi/strimzi-kafka-operator ***@***.***>
Cc: Eric V ***@***.***>, Author ***@***.***>
Subject: Re: [strimzi/strimzi-kafka-operator] Cannot upgrade using Helm (Discussion #8014)

 

Check out https://cloud-native.slack.com/archives/CMH3Q3SNP/p1675088509172959 ... I'm not a Helm user, so no idea what other options are there to get Helm to recognize this file.

**Reply by scholzj** [reply]
Well ... it still sucks. Shame nobody caught this before the release. I opened a thread on Slack if we should do 0.33.1 to remove it to fix this issue.

**Comment by eric-aops** [comment]
😉

 

I think if you add to the Lease template the three notations it will stop complaining. 

 

I.e. change this:

metadata:

  name: strimzi-cluster-operator

  namespace: {{ .Release.Namespace }}

  labels:

    app: {{ template "strimzi.name" . }}

    chart: {{ template "strimzi.chart" . }}

    component: lease

    release: {{ .Release.Name }}

    heritage: {{ .Release.Service }}

 

to this:

metadata:

  name: strimzi-cluster-operator

  namespace: {{ .Release.Namespace }}

  annotations:

    meta.helm.sh/release-name: {{ .Release.Name }}

    meta.helm.sh/release-namespace: {{ .Release.Namespace }}

  labels:

    app.kubernetes.io/managed-by: Helm

    app: {{ template "strimzi.name" . }}

    chart: {{ template "strimzi.chart" . }}

    component: lease

    release: {{ .Release.Name }}

    heritage: {{ .Release.Service }}

 

I haven’t tested this, guessing when you do the first upgrade you might have to run with –force to replace on the cluster, or even patch manually. 

But all following upgrades should work.

 

Kind regards,

 

Eric Van Steenbergen

 

Phone: +356 7711 1767

Email: ***@***.***

 

From: Jakub Scholz ***@***.***>
Reply to: strimzi/strimzi-kafka-operator ***@***.***>
Date: Thursday, 2 February 2023 at 17:25
To: strimzi/strimzi-kafka-operator ***@***.***>
Cc: Eric V ***@***.***>, Author ***@***.***>
Subject: Re: [strimzi/strimzi-kafka-operator] Cannot upgrade using Helm (Discussion #8014)

 

Well ... it still sucks. Shame nobody caught this before the release. I opened a thread on Slack if we should do 0.33.1 to remove it to fix this issue.
